### PR TITLE
docs: Update readme to show the current backstage version

### PR DESCRIPTION
**Pull request by SamoKopecky**
- Add a workflow that will create a new tag based on the changes to `backstage.json`
- Update some old information about the service catalog in README
- Reference the correct repository in the README header

### CODE REVIEW

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -45,3 +45,37 @@ jobs:
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            backstage-bump:
+              - 'backstage.json'
```
Comment (tumido) [inline .github/workflows/push.yaml:62]: Can you make this a separate workflow with the GH native path filter?

**Review by tumido — CHANGES_REQUESTED** [review]

File: .github/workflows/push.yaml (hunk 1 above)
Comment (SamoKopecky) [reply .github/workflows/push.yaml:62]: Will do.

Comment (SamoKopecky) [reply .github/workflows/push.yaml]: Looks much cleaner now, changed in the latest cmmit.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**